edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/ConnectionManager.cs

```diff
@@ -280,6 +277,23 @@ await device.CloudConnection.Filter(cp => cp.IsActive)
             }
         }
 
+        async Task CloseAllCloudConnections()
+        {
+            using (await this.connectToCloudLock.LockAsync())
```

Comment:
this lock is used to protect which resources/properties? this.devices?  If so, there are some other usages of this.devices in this class, should locking be required for other usages?

Reply:
Also, what deviceConnLock is used for? it looks like it also protect access to this.devices.

Reply:
The lock protects that there is no new connection open to the cloud while a disconnect from cloud is handled (closing all cloud connection). This is because of how the sdk works with AMQP multiplexing, in order to drop the tcp connection and create a new one there should be no existing open connection. 
deviceConnLock protects that there are no more ConnectedDevice created at the same time so it would exceed the limit, but it doesn't connect to cloud.